Add Leaky RELU kernel implementation

abhay-lal opened the pull request:
- Implement Leaky RELU kernel in aie2p (leaky_relu.cc)
- Add Leaky RELU example with Python API, golden reference, and host code
- Update README.md to mark Leaky RELU as completed (🟢)
- Add CMakeLists.txt entries for kernel and example

Leaky RELU implements f(x) = max(x, alpha * x) where alpha defaults to 0.01. The implementation follows the same pattern as the existing RELU kernel but adds support for the alpha parameter.

<!--
SPDX-FileCopyrightText: Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
SPDX-License-Identifier: Apache-2.0
-->

Describe the intent of your PR here.

## Added
-

## Changed
-

## Removed
-

## PR Merge Checklist

1. [x] The PR is rebased on the latest `devel` commit and pointing to `devel`.
2. [ ] Your PR reviewed and approved.
3. [ ] All checks are passing.

abhay-lal committed:
Add Leaky RELU kernel implementation

- Implement Leaky RELU kernel in aie2p (leaky_relu.cc)
- Add Leaky RELU example with Python API, golden reference, and host code
- Update README.md to mark Leaky RELU as completed (🟢)
- Add CMakeLists.txt entries for kernel and example

Leaky RELU implements f(x) = max(x, alpha * x) where alpha defaults to 0.01.
The implementation follows the same pattern as the existing RELU kernel
but adds support for the alpha parameter.

Victor-Jung commented:
Hi @abhay-lal, thanks for your contribution! I ran the [linting CI](https://github.com/amd/IRON/actions/runs/19179364865/job/54834100329?pr=2) and it failed. Check [the contributing guidelines](https://github.com/amd/IRON/blob/devel/CONTRIBUTING.md) to find out which commands to run (spoiler, use `black` for python and `clang-format-wrapper` for C++).

Victor-Jung reviewed (dismissed):
Apart from linting the code LGTM :)

abhay-lal committed:
Fix C++ formatting in leaky_relu.cc

- Split long function signature across multiple lines
- Remove trailing newline at end of file

abhay-lal committed:
Fix Python formatting with black

- Split long line in leaky_relu_golden.py
- Remove trailing newlines from both Python files

abhay-lal committed:
Fix C++ formatting in leaky_relu.cpp

- Remove trailing newline at end of file

abhay-lal committed:
Fix C++ formatting: ensure newline at end of file
